lib/datasource/cargo/index.js

```diff
@@ -5,30 +5,50 @@ module.exports = {
 };
 
 async function getPkgReleases({ lookupName }) {
-  const crateUrl = `https://crates.io/api/v1/crates/${lookupName}`;
+  if (!lookupName) {
+    return null;
+  }
+  const len = lookupName.length;
+  let path;
+  if (len === 1) {
+    path = '1/' + lookupName;
+  } else if (len === 2) {
+    path = '2/' + lookupName;
+  } else if (len === 3) {
+    path = '3/' + lookupName[0] + '/' + lookupName;
+  } else {
+    path =
+      lookupName.slice(0, 2) + '/' + lookupName.slice(2, 4) + '/' + lookupName;
+  }
+  const baseUrl =
+    'https://raw.githubusercontent.com/rust-lang/crates.io-index/master/';
+  const crateUrl = baseUrl + path;
   try {
-    const res = (await got(crateUrl, {
-      json: true,
+    let res = (await got(crateUrl, {
       platform: 'cargo',
     })).body;
-    if (!(res && res.crate && res.crate.name && res.versions)) {
+    res = res.split('\n');
+    res.pop(); // Pop empty last element (caused by trailing \n)
+    res = '[' + res.join(',') + ']';
```

Comment:
Instead of joining and then parsing, you could so `.map(JSON.parse)` in one step and have an array of objects.